framework/sandstone_tests.cpp

```diff
@@ -0,0 +1,102 @@
+#include <fnmatch.h>
+#include "sandstone_tests.h"
+
+static std::map<const char *, struct test *, SandstoneTestSet::cstr_cmp> all_test_map; /* maps test name to test */
+static std::map<const char *, std::vector <struct test *>> all_group_map; /* maps group name to a vector of tests it contains */
```

Comment:
Please don't use `std::map` globals (they have an init function). Move these to inside `SandstoneApplication`.

Use `std::string_view` as the key type instead of `const char *`, so you don't have to have `cstr_cmp`.